Makefile

```diff
@@ -13,23 +13,23 @@ CMAKEOBJ := CMakeLists.txt
 SUBDIR := test
 
 .DEFAULT_GOAL: runGame
-.PHONY: cppcheck cppcheckup valgrind diff doxygen buildunittest unittest
-all: runGame cppcheck cppcheckup valgrind diff
+.PHONY: cppcheck cppcheckup valgrind diff doc buildunittest unittest
+tests: runGame cppcheck cppcheckup valgrind diff
 
 runGame: $(OBJS)
 	$(RUN) $(CFLAGS) -o runGame $(OBJS)
 
 Parser.o: Parser.cpp
-	$(RUN) $(CFLAGS) -c Parser.cpp
+	$(RUN) $(CFLAGS) -c Parser.cpp Parser.h
 
 Game.o: Game.cpp
```

Comment:
Es a Game.h nem rantja be mondjuk a Unit.h-t? ;-) Mert akkor eleg, ha az megvaltozik.

Reply:
Közben letesztelgettem mindent. Nekem úgy tűnik, hogy csak akkor látja a make a változást, hogyha dependency-ként vannak a header fájlok is megadva. Hiába módosul a Unit.h, ha semelyik targetnél nincsen figyelve, hogy módosítva lett-e. Most hozzárendeltem minden headert az adott .o-s targethez, így minden módosításnál generál egy új object fájlt.

Reply:
Bezony, a make nem a makrok vegrehajtasa utani allapoto nezi. Ugye rekurzivan hozzaadtad a headereket? :-)